### DISCUSSION TITLE
Units make foods plural

### DISCUSSION THREAD

**Discussion by hjk321** [mealie-recipes/mealie#3609] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Consider the following two ingredients using the "ingredient amounts" feature:

- 1 tomato, diced 
- 1 cup tomatoes, diced

However, with the current behavior, the second one would print as "1 cup tomato, diced" because the system only checks the number when deciding to make a food plural. This can cause incorrect grammar. This also occurs with fractional amounts less than 1.

### Please provide a concise description of the feature that would resolve your issue.

I propose that the manage data screen for units have a "makes food plural" checkbox that is checked by default (as most units make food plural). This would affect how each ingredient in the list is printed.

This would be the behavior in a recipe:

1. If no unit is specified (eg 1 tomato), the current behavior applies and the number determines the plurality of the food.
2. If the specified unit does not have "makes food plural" checked, the current behavior also applies.
3. However, if the specified unit has this setting checked, the food will always display as plural.
4. The unit itself is not affected by this setting. "1/2 cup" would still use "cup" instead of "cups", for instance.
5. Obviously, if a food doesn't have a plural form, the standard name of the food is always used. 

### Please consider and list out some caveats or tradeoffs made in your design decision

The "manage data" screen is rarely accessed by a user so I don't think an extra checkbox would be a UX concern. I also think the extra boolean would pose no significant impact on database size or app performance.

If we decide it's better to leave "makes food plural" unchecked by default, that's fine too, but I'm pretty sure all of the seeded (default) units grammatically benefit from this setting checked.

Please note I'm only an English speaker, I don't know how useful this is for other languages.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
I agree with this, the current implementation leaves room to be desired.

I think it will need to be a little more configurable, though. Consider:
- 2 lbs chicken breast
- 2 chicken breasts

So units will also need to force foods to be singular sometimes. So we would need to add a field to units, maybe called plural handling:
- use default behavior
- make food singular
- make food plural

I don't know where the line is where we get too complicated/configurable, but I don't think we're there yet.

**Reply by hjk321** [reply]
That's a really good point. However it feels like only *certain* foods are forced plural/singular by *certain* units, for instance in your example lbs forces singular but for many foods lbs would make the food plural. So maybe we need to add these overrides to the individual foods? This is starting to get complicated...

**Reply by michael-genson** [reply]
Yeah I wouldn't want to go that far. Honestly just keeping it simple with putting "force plural" on the unit is probably best. The only area I'm not certain is other languages, though that's another can of worms

**Reply by logical-intent** [reply]
Have just discovered Mealie and my inner grammar nazi is bristling at "3 tablespoons milks".  I'm sure that only one kind of milk would be required!  

Mealie seems like a great app that a lot of work and care has gone into, so I'm not going to complain at all beyond adding this little note.

**Reply by zrieprakis** [reply]
On a basic level, if units are plural, ingredient doesn't have to be.  That isn't perfect, but at least fixes problems & doesn't cause any.  On a higher level, have connected AI correct the grammar for you.

**Reply by underspiffy** [reply]
I would love to see some more customization for pluralization as well. I think on a basic level, @zrieprakis has the right idea, just don't pluralize it if it has units, if it doesn't have a unit then pluralize it. Right now I find more things are incorrect than correct.

**Reply by michael-genson** [reply]
I have [a PR open](https://github.com/mealie-recipes/mealie/pull/6057) for this but I'm not happy enough with it to get it merged yet.
